### PR TITLE
Create list loader

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Add support for errors in create list account

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots
Screenshot 2022-02-01 at 14 56 56

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/ListModal/useStyles.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ const useStyles = makeStyles(({ typography }) => ({
   editButton: {
     color: "red",
   },
+  error: {
+    color: "red",
+  },
```
Comment (kilemensi) [inline src/components/ListModal/useStyles.js:18]: Any reason we're not using `palette.primary.main` here?

File: src/components/Lists/index.js [hunk 2]
```diff
@@ -44,6 +46,9 @@ function Lists({ results: listsProp, paginationProps, ...props }) {
   useEffect(() => {
     if (data) {
       setLists(data.results);
+      setLoading(true);
+    } else if (!data) {
+      setLoading(false);
```
Comment (kilemensi) [inline src/components/Lists/index.js:51]: Check both `data` and `error` on results of `useSWR` to be sure if we're in a loading state. [Docs](https://swr.vercel.app/docs/getting-started#make-it-reusable).

File: src/components/Lists/index.js [hunk 3]
```diff
@@ -112,6 +122,11 @@ function Lists({ results: listsProp, paginationProps, ...props }) {
           accountsLabel="User Accounts"
           accountsOnChange={handleChange}
           accountsHelper="Enter twitter account names seperated by a comma i.e userone,usertwo"
+          accountsErrorHelper={
+            !isLoading && isError
+              ? "Please enter a valid twitter account/s and that multiple accounts are separated by a comma."
```
Comment (kilemensi) [inline src/components/Lists/index.js:127]:
```suggestion
              ? "Please enter a valid comma-separated list of Twitter usernames."
```
Or something like that...

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: src/components/Lists/index.js [hunk 4]
```diff
@@ -125,20 +134,23 @@ function Lists({
...
       accounts: accountsMap,
     };
-
     try {
       await fetchJson("/api/lists", null, {
         method: "POST",
         body: JSON.stringify(payload),
       });
 
+      setLoading(true);
```
Comment (KhadijaMahanga) [inline src/components/Lists/index.js:143]: I think this should come at the very top of the function.. before the await function

**Review by KhadijaMahanga — COMMENTED** [review]
👍🏾

**Comment by KhadijaMahanga** [comment]
@nerdlet there's a small bug once you successful create a list, you don't see it unless you refresh (mutate fails). I think its because of this [line](https://github.com/CodeForAfrica/TwoopsTrackerUI/blob/6074430345cad8b9de56da1eee05e62db395dd77/src/components/Lists/index.js#L92) which return null. 
```
  const shouldFetch = () => {
    if (!paginating) {
      return null;
    }
```
I am not sure why we need paginating state... but if we must have it then we need to add a creating state and have something like `!(paginating || creating)`

File: src/components/Lists/index.js [hunk 5]
```diff
@@ -107,13 +109,20 @@ function Lists({
...
+
   useEffect(() => {
     setPaginating(false);
     if (data) {
       setLists({ ...data });
+      setLoading(true);
+    } else if (!data && !error) {
+      setLoading(true);
+    } else {
+      setLoading(false);
```
[the remark this answers is not in the document]
Comment (nerdlet) [reply src/components/Lists/index.js]: @KhadijaMahanga My bad, it should be `setLoading(false)`

**Comment by nerdlet** [comment]
> @nerdlet there's a small bug once you successful create a list, you don't see it unless you refresh (mutate fails). I think its because of this [line](https://github.com/CodeForAfrica/TwoopsTrackerUI/blob/6074430345cad8b9de56da1eee05e62db395dd77/src/components/Lists/index.js#L92) which returns null if paginating is false.
> 
> ```
>   const shouldFetch = () => {
>     if (!paginating) {
>       return null;
>     }
> ```
> 
> I am not sure why we need paginating state... but if we must have it then we need to add a creating state and have something like `!(paginating || creating)`

@KhadijaMahanga Same when deleting the list, it doesn't work until page is refreshed. I think we need the pagination  state for pagination component.  The only other significant change that could affect this would be the `listProp`.

**Review by KhadijaMahanga — APPROVED** [review]
🚀

**Pull request merged**